typing/typetexp.mli

```diff
@@ -17,23 +17,32 @@
 
 open Types
 
+module TyVarEnv : sig
+  val reset : unit -> unit
+  (* removes all type variables from scope *)
+
+  val narrow_in: (unit -> 'a) -> 'a
+  (* Evaluate in a narrowed type-variable scope *)
+
+  type poly_univars
+  val make_poly_univars : string list -> poly_univars
+    (* Create a set of univars with given names *)
+  val check_poly_univars :
+     Env.t -> Location.t -> poly_univars -> type_expr list
+    (* Verify that the given univars are universally quantified,
+       and return the list of variables. The type in which the
+       univars are used must be generalised *)
+end
```

Comment:
I don't really understand how you chose which things to put in this module

Reply:
These happen to be the functions used externally. No other logic here. I've added a small comment to answer your reasonable question.